src/ui/public/state_management/state.js

```diff
@@ -80,13 +80,16 @@ export default function StateProvider(Private, $rootScope, $location, config) {
       $location.search(search).replace();
     }
 
-    if (risonEncoded) {
+    if (!risonEncoded) {
+      return null;
+    }
+
+    if (this.isHashingEnabled()) {
```

Comment:
Because there are so many conditionals and early-exits in this method, I think it's worth adding a comment describing the state of the app in which this condition is reached:
```javascript
if (this.isHashingEnabled()) {
  // If Kibana has been loaded for the first time with RISON states URL, and the user has has hashed states
  // enabled, then update the URL to show the hashed states instead of the RISON states.
}
```